pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -304,6 +318,54 @@ def _register_ddp_hooks(self) -> None:
                 ddp_comm_wrapper=self._ddp_comm_wrapper,
             )
 
+            # Post-localSDG is only available after 1.9,
+            # and `torch.distributed.optim` package currently is not available on Windows.
+            if (
+                not _IS_WINDOWS
+                and _TORCH_GREATER_EQUAL_1_9
+                and isinstance(self._ddp_comm_state, post_localSGD.PostLocalSGDState)
+                and self.lightning_module.trainer.state.fn == TrainerFn.FITTING
+            ):
+                self._reinit_optimizers_with_post_localSGD(self._ddp_comm_state.start_localSGD_iter)
+
+    def _reinit_optimizers_with_post_localSGD(self, warmup_steps: int):
+        optimizers = self.lightning_module.trainer.optimizers
+        if self._model_averaging_period is None:
+            rank_zero_warn(
+                "Post-localSGD algorithm is used, "
+                "but model averaging period is not provided to DDP plugin. Set this period as 16."
+            )
+            self._model_averaging_period = 16
```

Comment:
instead of defaulting to 16, should we assert that a value for model averaging period is set if `PostLocalSGDState` is used? and raise a misconfiguration exception if not?

Reply:
Sounds good.